source/Ship.cpp

```diff
@@ -1300,9 +1300,37 @@ void Ship::Place(Point position, Point velocity, Angle angle, bool isDeparting)
 				bay.ship->SetSwizzle(bay.ship->customSwizzle >= 0 ? bay.ship->customSwizzle : swizzle);
 		}
 	}
+	    // Get the weapon ranges for this ship, so the Ai can call it.
+    const vector<Hardpoint> &hardpoints = armament.Get();
+    for(unsigned i = 0; i < hardpoints.size(); ++i)
+    {
+			const Weapon *weapon = hardpoints[i].GetOutfit();
+			if(weapon && !weapon->Ammo() && weapon->DoesDamage() && hardpoints[i].IsTurret())
+				turretRange = max(turretRange, weapon->Range() + hardpoints[i].GetPoint().Length());
+			else if(weapon && !weapon->Ammo() && weapon->DoesDamage() && hardpoints[i].IsGun())
+				gunRange = max(gunRange, weapon->Range() + hardpoints[i].GetPoint().Length());
+			// if(weapon && !weapon->Ammo() && weapon->DoesDamage() && hardpoints[i].IsDefensive())
+				// defenseRange = max(defenseRange, weapon->Range() + hardpoints[i].GetPoint().Length());
```

Comment:
```suggestion
		const Weapon *weapon = hardpoints[i].GetOutfit();
		if(weapon && !weapon->Ammo() && weapon->DoesDamage() && hardpoints[i].IsTurret())
			turretRange = max(turretRange, weapon->Range() + hardpoints[i].GetPoint().Length());
		else if(weapon && !weapon->Ammo() && weapon->DoesDamage() && hardpoints[i].IsGun())
			gunRange = max(gunRange, weapon->Range() + hardpoints[i].GetPoint().Length());
```
some identitation went wrong here XD